src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java

```diff
@@ -400,9 +401,9 @@ public void addEntriesToSnapshot(
           batchRetrieveById(datasetFirestore, datasetId, batch);
 
       // Find directory paths that need to be created; plus add to the cache
-      List<String> newPaths = FileMetadataUtils.findNewDirectoryPaths(datasetEntries, pathMap);
+      Set<String> newPaths = FileMetadataUtils.findNewDirectoryPaths(datasetEntries, pathMap);
       List<FireStoreDirectoryEntry> datasetDirectoryEntries =
-          batchRetrieveByPath(datasetFirestore, datasetId, newPaths);
+          batchRetrieveByPath(datasetFirestore, datasetId, new ArrayList<>(newPaths));
```

Comment:
@pshapiro4broad  - This is the one place where I think we still want a list. Is this the best way to convert from set to list?

Reply:
This is OK if you need the result to be modifiable. If the result can be read-only (which appears to be true here), I would use `List.copyOf()` instead:
```suggestion
          batchRetrieveByPath(datasetFirestore, datasetId, List.copyOf(newPaths));
```